### ISSUE TITLE
Another mutation type issue

### ISSUE THREAD

**Issue by atoy40** [trojanowski/react-apollo-hooks#25]
When you use the "update"property in mutation options like :

```javascript
const myMut = useMutation<MyMut, MyMutVars>(MYMUT, {
    update: (cache, result) => {
        //// result has type FetchResult<MyMut>
        //// result.data has type ExecutionResult<ExecutionResultDataDefault> instead of an expected ExecutionResult<MyMut>
    }
});
```

I think the problem comes again from the FetchResult type as defined in apollo-link. The only workarround I think is to redefine MutationOptions/MutationUpdaterFn type to use the FetchResult type you already redefine. :s

**Comment by trojanowski** [comment]
@atoy40 Could you please check if #26 solves the problem for you?

**Comment by atoy40** [comment]
> @atoy40 Could you please check if #26 solves the problem for you?

I'll try it tomorrow, but it looks good :)

**Comment by atoy40** [comment]
Perfect ! :
![update-type](https://user-images.githubusercontent.com/229339/49217484-f6d80e80-f3cd-11e8-9eba-11b8acefac82.png)

**Comment by trojanowski** [comment]
I published a version 0.1.8 with the fix.
